mlrun/feature_store/api.py

```diff
@@ -292,6 +292,13 @@ def infer(
 
     namespace = namespace or get_caller_globals()
     if featureset.spec.require_processing():
+        _, default_final_state, _ = featureset.graph.check_and_process_graph(
+            allow_empty=True
+        )
+        if not default_final_state:
+            raise mlrun.errors.MLRunPreconditionFailedError(
```

Comment:
i assume we still need to force the user to have default final state?

Reply:
Forcing in addition to this exception?